File: aula3/exercicio2.py
#!/usr/local/bioinformatic/conda3/bin/python

seqDNA = input("Digite sua sequencia de DNA\n")
seqDNA = seqDNA.upper()

tamanho = len(seqDNA)
#print(tamanho)

revDNA=[]
#revDNA.append("A")
#print(revDNA)

dnaLetters=set("ATCG")

if(set(seqDNA).issubset(dnaLetters)):
    print(f"Sequencia de DNA = {seqDNA}")
    print("Reverso Complementar = ", end="")
    for i in range((tamanho-1),-1,-1):
        #print(seqDNA[i])
        if (seqDNA[i] =="A"):
            revDNA.append("T")
        elif(seqDNA[i] == "G"):
            revDNA.append("C")
        elif(seqDNA[i] == "C"):
            revDNA.append("G")
        else:
            revDNA.append("A")

    print("".join(revDNA))
    print("Sequencia de invertida com sucesso")
else:
    print("Sequencia possui caracteres invalidos para ser DNA")

File: aula4/exercicio2.R
#!/usr/local/bin/Rscript

cat("Digite sua sequencia de DNA\n")
seqDNA <- readLines(con = "stdin", n = 1)
seqDNA <- toupper(seqDNA)

tamanho <- nchar(seqDNA)
revDNA <- c()
dnaLetters <- c("A", "T", "C", "G")
seq_chars <- strsplit(seqDNA, NULL)[[1]]

if (all(seq_chars %in% dnaLetters)) {
  cat(sprintf("Sequencia de DNA = %s\n", seqDNA))
  cat("Reverso Complementar = ")
  
  for (i in tamanho:1) {
    nt <- seq_chars[i]
    if (nt == "A") {
      revDNA <- c(revDNA, "T")
    } else if (nt == "G") {
      revDNA <- c(revDNA, "C")
    } else if (nt == "C") {
      revDNA <- c(revDNA, "G")
    } else {
      revDNA <- c(revDNA, "A")
    }
  }
  
  cat(paste0(revDNA, collapse = ""), "\n")
  cat("Sequencia de invertida com sucesso\n")
} else {
  cat("Sequencia possui caracteres invalidos para ser DNA\n")
}